src/ui-kit/theme/ThemeProvider.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useMemo, useState } from 'react';
+import { ThemeProvider } from '@mui/material';
+import { createCommonTheme, palette } from './theme';
+
+export const ThemeContext = createContext();
+
+export const CombinedThemeProvider = ({ children }) => {
+  const [currentTheme, setCurrentTheme] = useState('light');
```

Comment:
The initial value should be taken from localStorage, if it is not, then set 'light'